src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcTaskHost.cs

```diff
@@ -484,10 +484,7 @@ internal bool AcquireAndSetUpHost(HandshakeOptions hostContext, INodePacketFacto
         /// </summary>
         internal void DisconnectFromHost(HandshakeOptions hostContext)
         {
-            ErrorUtilities.VerifyThrow(_nodeIdToPacketFactory.ContainsKey((int)hostContext) && _nodeIdToPacketHandler.ContainsKey((int)hostContext), "Why are we trying to disconnect from a context that we already disconnected from?  Did we call DisconnectFromHost twice?");
-
-            _nodeIdToPacketFactory.Remove((int)hostContext);
```

Comment:
This change technically means removing hostContext isn't atomic, but I don't think that's an issue.

Reply:
I would slightly prefer the actual logic to be on its own lines and not part the `VerifyThrow()` call. It makes it easier to read for me when error handling does not mutate the state.